spec/features/categories/index_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+RSpec.describe 'Testing Category#index view', type: :feature do
+  before(:each) do
+    # Creating a test user
+    @user = User.create(name: 'Test User', email: 'example@test.com', password: '123456')
+
+    # Creating test categories for the user
+    @categories = [
+      Category.create(name: 'Category 1', icon: 'icon1', user: @user),
+      Category.create(name: 'Category 2', icon: 'icon2', user: @user)
+    ]
+
+    # Creating test expenses for each category
+    Expense.create(category: @categories[0], amount: 100, created_at: Time.now - 1.day)
+    Expense.create(category: @categories[0], amount: 150, created_at: Time.now)
+    Expense.create(category: @categories[1], amount: 200, created_at: Time.now)
+  end
+
+  it 'renders the category list with correct information' do
+    # Visiting the user's category index page
+    visit user_categories_path(@user)
+
+    # Assertions for elements on the page
+    expect(page).to have_selector('header h2', text: 'CATEGORIES')
+    expect(page).to have_selector('header i.fa.fa-search')
+    expect(page).to have_selector('.category-list')
+
+    # Looping through each category and making assertions
+    @categories.each do |category|
+      expect(page).to have_selector('.category-link', count: @categories.length)
+
+      # Using within to scope the assertions to the specific category link
+      within(".category-link[href='#{user_category_expenses_path(@user, category)}']") do
+        expect(page).to have_selector(".category-icon img[src=\"#{category.icon}\"]")
+        expect(page).to have_selector('.category p', text: category.name)
+        expect(page).to have_selector('.category p.expense-date', text: category.created_at.strftime('%m/%d/%Y'))
+        expect(page).to have_selector('.amount p', text: number_to_currency(category.expenses.sum(:amount)))
```

Comment:
- [ ] Please kindly take note that this test is failing, though I observed that it passed in the previous commit before the review. Kindly re-adjust the implementation; I think the reason for this is the manipulation you did to display the total amount for individual category. 🙏